### PR TITLE
Ajsharm/detector author publish flow

### PR DESCRIPTION
**Pull request by SharmaAjay19**
Applens UI and Applens backend changes to add utterances (Search terms) to detectors

### CODE REVIEW

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 1]
```diff
@@ -6,10 +6,71 @@ <h4>Confirm?</h4>
...
+        <span>Make your detector discoverable for users by adding at least 5 search terms that describe the problems your detector can solve. We have some <strong>recommendations based on the detector description you provided</strong>.</span>
+        <table style="width:100%;table-layout:fixed;">
+          <tr>
+            <td><div class="utterancestab"><strong>Add new Search Terms from below</strong></div></td>
+            <td><div class="utterancestab"><strong>Search terms already added to your detector</strong></div></td>
+          </tr>
+          <tr>
+            <td class="utterancessection">
+              <div class="utterancestabcontent">
+                <div style="margin-left: 40px; margin-top: 10px;">
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html]: move style elements to css class

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 2]
```diff
@@ -258,13 +264,17 @@ export class OnboardingFlowComponent implements OnInit {
       })
       .subscribe((response: any) => {
         this.queryResponse = response.body;
+        if (this.queryResponse.invocationOutput.suggestedUtterances) {
+          this.recommendedUtterances = this.queryResponse.invocationOutput.suggestedUtterances.results;
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts]: Can the results be undefined and if yes, the below code where you are doing indexOf will fail

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts [hunk 3]
```diff
@@ -308,21 +320,53 @@ export class OnboardingFlowComponent implements OnInit {
     }
   }
 
+  addUtterance(utterance: RecommendedUtterance) {
+    var index = this.allUtterances.indexOf(utterance.sampleUtterance);
+    if (index<0) {
```
Comment (ShekharGupta1988) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.ts]: just format the file in VSCode edit

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 4]
```diff
@@ -6,10 +6,71 @@ <h4>Confirm?</h4>
       <hr style="margin-top: 10px;margin-bottom: 10px;">
       <span>Published changes will be live in Applens and App Services "Diagnose and Solve" blade for external
         customers if the detector is marked public.</span>
-
-      <div style="margin-top:10px;margin-bottom: 10px">
+      <div style="width:100%;height:450px">
```
Comment (dagrooms52) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html]: Was this tested on Medium and Small screen sizes? 450px is significant height on small screens and is not adaptive, I want to be sure the layout isn't affected

Comment (dagrooms52) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html]: As a whole this should be a separate Angular component, it's a lot of code and is going to take over the onboarding-flow code

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.scss [hunk 5]
```diff
@@ -291,3 +291,131 @@ span.rotate {
     visibility: visible;
     opacity: 1;
 }
+
+.utterancestab {
```
Comment (dagrooms52) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.scss]: Are you using VS Code + Editorconfig plugin? It should automatically set your indent to 4 spaces in Sass files

File: AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html [hunk 6]
```diff
@@ -6,10 +6,71 @@ <h4>Confirm?</h4>
...
+                      <i class="fa fa-trash utterancesremoveicon" (click)="removeUtterance(utterance)"></i>
+                    </li>
+                  </ul>
+                </div>
+              </div>
+            </td>
+          </tr>
+        </table>
+      </div>
+      <div style="margin-top:10px;margin-bottom: 10px;position: relative;">
```
Comment (cindylovescoding) [inline AngularApp/projects/applens/src/app/modules/dashboard/onboarding-flow/onboarding-flow.component.html]: Could you move all these styles into scss files?

**Pull request merged**